contrib/fsl_integration/sct_plugin.py

```diff
@@ -208,52 +290,703 @@ def __init__(self, parent):
+    (2018)</i>. BMJ Open. 13;8(4)e01980
+    """
+
+
+    def __init__(self, parent):
+        super(TabPanelCSA, self).__init__(parent=parent,
+                                          id_=wx.ID_ANY)
+        button_gm = wx.Button(self, id=wx.ID_ANY, label="Process Segmentation")
+        button_gm.Bind(wx.EVT_BUTTON, self.onButtonCSA, id = button_gm.GetId())
+
+        lbl_vertfile = ['Off', 'On']
+        self.rbox_vertfile = wx.RadioBox(self, label='Vertfile:',
+                                         choices=lbl_vertfile,
+                                         majorDimension=1,
+                                         style=wx.RA_SPECIFY_ROWS)
+
+        lbl_perlevel = ['No', 'Yes']
+        self.rbox_perlevel = wx.RadioBox(self, label='Perlevel:',
+                                         choices=lbl_perlevel,
+                                         majorDimension=1,
+                                         style=wx.RA_SPECIFY_ROWS)
+
+        hbox1 = wx.BoxSizer(wx.HORIZONTAL)
+        l1 = wx.StaticText(self, wx.ID_ANY, "Vertebral Levels/Slice Range")
+        hbox1.Add(l1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+        self.t1 = wx.TextCtrl(self)
+        self.t1.Bind(wx.EVT_TEXT, self.OnKeyTyped)
+        hbox1.Add(self.t1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+
+        hbox2 = wx.BoxSizer(wx.HORIZONTAL)
+        l2 = wx.StaticText(self, wx.ID_ANY, "Output Table Name")
+        hbox2.Add(l2, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+        self.t2 = wx.TextCtrl(self)
+        self.t2.Bind(wx.EVT_TEXT, self.OnKeyTyped)
+        hbox2.Add(self.t2, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(self.rbox_perlevel, 0, wx.ALL, 5)
+        sizer.Add(hbox1)
+        sizer.Add(hbox2)
+        sizer.Add(self.rbox_vertfile, 0, wx.ALL, 5)
+        sizer.Add(button_gm, 0, wx.ALL, 5)
+        self.sizer_h.Add(sizer)
+        self.SetSizerAndFit(self.sizer_h)
+
+    def OnKeyTyped(self, event):
+        txt = event.GetString()
+
+
+    def onButtonCSA(self, event):
+
+        overlayORD = displayCtx.overlayOrder
+
+        print('Overlay Order:', overlayORD)
+
+        if len(overlayORD) > 1:
+
+            img2 = overlayORD[1]
+            segimg = overlayList[img2].dataSource
+
+        else:
+
+            img2 = overlayORD[0]
+            segimg = overlayList[img2].dataSource
+
+        print('Segmentation:', segimg)
+
+        vertfile = self.rbox_vertfile.GetStringSelection()
+        perlevel = self.rbox_perlevel.GetStringSelection()
+        vert = self.t1.GetValue()
+        tabname = self.t2.GetValue()
+
+        if vert == '':
+            print('Vertebral labeling or slice range not defined.')
+        else:
+            print('Verts:', vert)
+
+        if tabname == '':
+            print('Output table name not defined, standard name will be used.')
+        else:
+            print('Output Table Name:', tabname)
```

Comment:
indicate what will be the default

also: indicate that extension .csv will be automatically added